src/components/Results.tsx

```diff
@@ -117,6 +124,29 @@ export const Results: React.FC<{
     return null;
   }
 
+  const searchResultsRefs = useMemo(
+    () =>
+      filteredFacilities.reduce((refs, facility) => {
+        const ref = createRef<HTMLDivElement>();
+        return { ...refs, [facility.id]: ref };
+      }, {}),
+    []
```

Comment:
I guess `filteredFacilities` must be added to the dependencies of this memoised constant.

Reply:
Yes, you are right. Thank you. Will update this